conversions/test/convertTest.cpp

```diff
@@ -0,0 +1,84 @@
+#include <gtest/gtest.h>
+#include <stdio.h>
+#include <stdlib.h>
+#include <math.h>
+#include <cstring>
+#include <mrpt/obs/CObservation2DRangeScan.h>
+#include <mrpt/obs/CObservationOdometry.h>
+#include <mrpt/poses/CPose3D.h>
+#include "mrpt_bridge/vrep_conversion.h"
+using namespace mrpt::obs;
+using namespace mrpt::poses;
+using namespace vrep_bridge;
+
+/**
+*This unit test tests the CObservation2DRangeScan conversion method. It tests the following parameters of the object created with the functions parameters.
+*- <b>obj.aperture[]</b> -> scanningAngle
+*- <b>obj.maxRange</b> -> maxScanDistance
+*- <b>obj.sensorPose</b> -> sensor_pose
+*
+*/
+TEST(ConvertTest, CObservation2DRangeScanTest)
+{
```

Comment:
You should be checking the actual data of the laser scan.

Grab an actual simulated laser scan from V-REP..

Write its contents to an e.g., textfile (or just dump them in the cpp file for now. Then in this test function read them back, convert them to a CObservation2DRangeScan and check that the data of the latter object is still there as it should be...

This may require some tinkering with capturing the data and reading /writing that text file but it is the most definitive way moving forward to make sure that the convert method produces the expected results...

Refer to this as an example: https://github.com/MRPT/mrpt/blob/master/samples/slam_icp_simple_example/test.cpp

Let me know if you need a more detailed explanation on this